src/libvirtApi/domain.ts

```diff
@@ -139,8 +139,8 @@ export interface DiskSpec {
     type: string,
     file?: string,
     device: string,
-    poolName: string,
-    volumeName: string,
+    poolName?: string | undefined,
+    volumeName?: string | undefined,
```

Comment:
? / undefined again?

Reply:
Yeah... I start to think we should make it a rule to always add "| undefined" when there is a "?". Otherwise it just gets annoying when you want to pass a value that might be undefined. There really shouldn't be any difference (in our APIs) between an object having a property with value undefined, and the object not having that property.